### PR TITLE
unblock kafka partitions

### PR DESCRIPTION
**Pull request by Vadman97**
## Summary

Tuning for kafka based on recent consumer issues.
* Broker crash was caused by large messages that would not be able to replicate because of the broker `replica.fetch.max.bytes` setting
* Consumer would not load large messages because of the max message size setting, resulting in partitions that were significantly behind.

## How did you test this change?

Deploying to production, monitoring kafka consumption.

## Are there any deployment considerations?

No

## Does this work require review from our design team?

No

### CODE REVIEW

File: backend/kafka-queue/kafkaqueue.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ const ConsumerGroupName = "group-default"
 const (
 	TaskRetries           = 2
 	prefetchQueueCapacity = 8
-	MaxMessageSizeBytes   = 16 * 1000 * 1000 // MB
+	MaxMessageSizeBytes   = 128 * 1024 * 1024 // MiB
```
Comment (Vadman97) [inline backend/kafka-queue/kafkaqueue.go:30]: align this with kafka broker setting

File: backend/kafka-queue/kafkaqueue.go [hunk 2]
```diff
@@ -101,7 +101,9 @@ func New(ctx context.Context, topic string, mode Mode, configOverride *ConfigOve
 	brokers := strings.Split(servers, ",")
 	groupID := strings.Join([]string{ConsumerGroupName, topic}, "_")
 
-	tlsConfig := &tls.Config{}
+	tlsConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
```
Comment (Vadman97) [inline backend/kafka-queue/kafkaqueue.go:105]: might help with some EOF errors we would get

File: backend/kafka-queue/kafkaqueue.go [hunk 3]
```diff
@@ -170,21 +172,31 @@ func New(ctx context.Context, topic string, mode Mode, configOverride *ConfigOve
 			Balancer:     &kafka.Hash{},
 			RequiredAcks: kafka.RequireOne,
 			Compression:  kafka.Zstd,
-			Async:        true,
+			// ensure messages are pushed to kafka before responding to the client
```
Comment (Vadman97) [inline backend/kafka-queue/kafkaqueue.go:175]: switch back to sync submit to have better client data sending guarantees

File: backend/kafka-queue/kafkaqueue.go [hunk 4]
```diff
@@ -170,21 +172,31 @@ func New(ctx context.Context, topic string, mode Mode, configOverride *ConfigOve
...
+			Logger: kafka.LoggerFunc(log.WithFields(log.Fields{
+				"code.module": "kafkaqueue",
+				"mode":        "producer",
+				"topic":       topic,
+			}).Debugf),
+			ErrorLogger: kafka.LoggerFunc(log.WithFields(log.Fields{
+				"code.module": "kafkaqueue",
+				"mode":        "producer",
+				"topic":       topic,
+			}).Errorf),
```
Comment (Vadman97) [inline backend/kafka-queue/kafkaqueue.go:192]: logger to help with tracking errors

File: backend/kafka-queue/kafkaqueue.go [hunk 5]
```diff
@@ -204,19 +216,33 @@ func New(ctx context.Context, topic string, mode Mode, configOverride *ConfigOve
 			Brokers:           brokers,
 			Dialer:            dialer,
 			HeartbeatInterval: time.Second,
+			ReadLagInterval:   time.Second,
 			SessionTimeout:    10 * time.Second,
 			RebalanceTimeout:  rebalanceTimeout,
-			ReadBatchTimeout:  KafkaOperationTimeout,
 			Topic:             pool.Topic,
 			GroupID:           pool.ConsumerGroup,
+			MinBytes:          1_000 * 1_000, // 1 MB
```
Comment (Vadman97) [inline backend/kafka-queue/kafkaqueue.go:224]: batch reads to reduce broker load

File: backend/kafka-queue/kafkaqueue.go [hunk 6]
```diff
@@ -204,19 +216,33 @@ func New(ctx context.Context, topic string, mode Mode, configOverride *ConfigOve
...
 			MaxBytes:          MaxMessageSizeBytes,
-			MaxWait:           time.Second,
+			MaxWait:           KafkaOperationTimeout,
+			ReadBatchTimeout:  KafkaOperationTimeout,
+			ReadBackoffMin:    time.Millisecond,
+			ReadBackoffMax:    5 * time.Second,
```
Comment (Vadman97) [inline backend/kafka-queue/kafkaqueue.go:229]: reduce minimum read delay

**Review by deltaepsilon — APPROVED** [review]

**Pull request merged**